### PR TITLE
Error when an invalid (as opposed to incorrect) unseal key is given.

### LINKED ISSUE #1777: vault unseal doesn't error out, and exits 0 when given an incorrect hex or base64 string
Using vault v0.6.1 (released version)

When giving an incorrect string for the unseal key (such as 'asdf'), there is no obvious visual output that an incorrect string has been entered, beyond the `Unseal Progress` staying at 0.  

This also exits 0, while other invalid keys given (such as `'key' must be a valid hex or base64 string`) exit non-zero.

```
-- sarah@vault:~ $ vault unseal
Key (will be hidden): <asdf>
Sealed: true
Key Shares: 6
Key Threshold: 2
Unseal Progress: 0
-- sarah@vault:~ $ echo $?
0
```

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #1777

### CODE REVIEW

File: http/sys_seal.go [hunk 1]
```diff
@@ -118,10 +118,18 @@ func handleSysUnseal(core *vault.Core) http.Handler {
 			if _, err := core.Unseal(key); err != nil {
 				// Ignore ErrInvalidKey because its a user error that we
 				// mask away. We just show them the seal status.
-				if !errwrap.ContainsType(err, new(vault.ErrInvalidKey)) {
+				switch {
```
Comment (vishalnayak) [inline http/sys_seal.go]: The above comment can now be removed.

**Comment by vishalnayak** [comment]
LGTM!

File: http/sys_seal.go (hunk 1 above)
Comment (jefferai) [inline http/sys_seal.go]: Good point.

Comment (jefferai) [inline http/sys_seal.go]: I've got some other stuff in my tree so I'll remove post-merge.

**Pull request merged**